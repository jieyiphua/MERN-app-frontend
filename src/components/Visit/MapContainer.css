.markercluster-map {
  height: 500px;
  width: 1000px;
  z-index: 0;
}
