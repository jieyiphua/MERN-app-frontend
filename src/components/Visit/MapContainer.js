import React from "react";
import { Map, TileLayer, Marker, Popup } from "react-leaflet";
import "./MapContainer.css";

class MapContainer extends React.Component {
  render() {
    const position = [-37.56138, 143.8752];
    return (
      <Map
        className="markercluster-map"
        center={position}
        zoom={15}
        minZoom={9}
        maxZoom={17}
      >
        <TileLayer
          url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
          attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
        />
        <Marker position={position}>
          <Popup>A pretty CSS3 popup. Easily customizable.</Popup>
        </Marker>
        <Marker position={position} />>
      </Map>
    );
  }
}

export default MapContainer;
