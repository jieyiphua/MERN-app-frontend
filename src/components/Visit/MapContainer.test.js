import MapContainer from "./MapContainer";
import React from "react";
import Enzyme, { shallow, render, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { createSerializer } from "enzyme-to-json";
import sinon from "sinon";

describe("<MapContainer />", () => {
  it("renders component", () => {
    const wrapper = shallow(<MapContainer />);
    expect(wrapper).toMatchSnapshot();
  });
});
