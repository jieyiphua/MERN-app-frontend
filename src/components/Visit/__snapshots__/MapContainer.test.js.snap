// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MapContainer /> renders component 1`] = `
<Map
  center={
    Array [
      -37.56138,
      143.8752,
    ]
  }
  className="markercluster-map"
  maxZoom={17}
  minZoom={9}
  zoom={15}
>
  <ForwardRef(Leaflet(TileLayer))
    attribution="Â© <a href=\\"http://osm.org/copyright\\">OpenStreetMap</a> contributors"
    url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
  />
  <ForwardRef(Leaflet(Marker))
    position={
      Array [
        -37.56138,
        143.8752,
      ]
    }
  >
    <ForwardRef(Leaflet(Popup))>
      A pretty CSS3 popup. Easily customizable.
    </ForwardRef(Leaflet(Popup))>
  </ForwardRef(Leaflet(Marker))>
  <ForwardRef(Leaflet(Marker))
    position={
      Array [
        -37.56138,
        143.8752,
      ]
    }
  />
  &gt;
</Map>
`;
